src/lib/schematics/update/update-schematic.md

```diff
@@ -0,0 +1,169 @@
+
+Also besides separating the upgrade data based on the target version, we split the upgrade data based on the type of code that is affected by these migrations:
+
+* See here: [src/lib/schematics/update/material/data](https://github.com/angular/material2/tree/master/src/lib/schematics/update/material/data)
+
+### Adding upgrade data
+
+Adding upgrade data is now an essential step before breaking changes should be merged into `upstream`.  For simple and common breaking changes, there should be already an upgrade data file that just needs the new change inserted.
+
+In case there is no upgrade data for a breaking change, we need to evaluate if there should be a single `misc` migration rule that is tied to that specific breaking change, or if we should create a new migration rule in a more generic way.
+
+---
+
+**Example**: Adding upgrade data for a property rename
+**Scenario**: In Angular Material V7.0.0, we rename `MatRipple#color` to `MatRipple#newColor`.
+
+First, look for an existing upgrade data file that covers similar breaking changes. In that case an existing upgrade data file for `property-names` already exists. Insert the new breaking change within the proper `VersionTarget`.
+
+_src/lib/schematics/update/material/data/property-names.ts_
+```ts
+export const propertyNames: VersionChanges<MaterialPropertyNameData> = {  
+  [TargetVersion.V7]: [  
+    {  
+      pr: '{PULL_REQUEST_LINK_FOR_BREAKING_CHANGE}',  
+      changes: [  
+        {  
+          replace: 'color',  
+          replaceWith: 'newColor',  
+          whitelist: {  
+            classes: ['MatRipple']  
+          }  
+        }  
+      ]  
+    }  
+  ],
+  
+  ...
+};
+```
+Once the data is inserted into the upgrade data file, the update schematic will properly migrate `MatRipple#color` to `MatRipple#newColor` if someone upgrades to Angular Material V7.0.0.
+
+But that's not all. It's encouraged to add a test-case for the new migration data. In this case, a test case already exists for the type of migration and we just need to add our breaking change to it. Read more about adding a test case in the next section.
+
+### Adding a breaking change to a test case
+
+Considering we added a breaking change to the update schematic, it's encouraged to add a proper test case for the new change that has been added. 
+
+In the scenario where a property from `MatRipple` has been renamed in V7, we don't need to create a new test-case file because there is already a test case for the `property-names` upgrade data. In that case, we just need to add the breaking change to the existing test case.
+
+_src/lib/schematics/update/test-cases/v7/property-names_input.ts_
+```ts
+...
+/**
+ * Mock definitions. This test case does not have access to @angular/material.
+ */ 
+
+class MatRipple {  
```

Comment:
I don't follow why this is here

Reply:
This is just an example of a test case. I wanted to use some real class in order to make it clear how upgrade data can be added, and how a test case can be created.